<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Strict//EN">
<html>
<head><title>RNAseq.fastq FastQC Report</title>

<style type="text/css">

 @media screen {
  div.summary {
    width: 18em;
    position:fixed;
    top: 3em;
    margin:1em 0 0 1em;
  }
  
  div.main {
    display:block;
    position:absolute;
    overflow:auto;
    height:auto;
    width:auto;
    top:4.5em;
    bottom:2.3em;
    left:18em;
    right:0;
    border-left: 1px solid #CCC;
    padding:0 0 0 1em;
    background-color: white;
    z-index:1;
  }
  
  div.header {
    background-color: #EEE;
    border:0;
    margin:0;
    padding: 0.5em;
    font-size: 200%;
    font-weight: bold;
    position:fixed;
    width:100%;
    top:0;
    left:0;
    z-index:2;
  }

  div.footer {
    background-color: #EEE;
    border:0;
    margin:0;
	padding:0.5em;
    height: 1.3em;
	overflow:hidden;
    font-size: 100%;
    font-weight: bold;
    position:fixed;
    bottom:0;
    width:100%;
    z-index:2;
  }
  
  img.indented {
    margin-left: 3em;
  }
 }
 
 @media print {
	img {
		max-width:100% !important;
		page-break-inside: avoid;
	}
	h2, h3 {
		page-break-after: avoid;
	}
	div.header {
      background-color: #FFF;
    }
	
 }
 
 body {    
  font-family: sans-serif;   
  color: #000;   
  background-color: #FFF;
  border: 0;
  margin: 0;
  padding: 0;
  }
  
  div.header {
  border:0;
  margin:0;
  padding: 0.5em;
  font-size: 200%;
  font-weight: bold;
  width:100%;
  }    
  
  #header_title {
  display:inline-block;
  float:left;
  clear:left;
  }
  #header_filename {
  display:inline-block;
  float:right;
  clear:right;
  font-size: 50%;
  margin-right:2em;
  text-align: right;
  }

  div.header h3 {
  font-size: 50%;
  margin-bottom: 0;
  }
  
  div.summary ul {
  padding-left:0;
  list-style-type:none;
  }
  
  div.summary ul li img {
  margin-bottom:-0.5em;
  margin-top:0.5em;
  }
	  
  div.main {
  background-color: white;
  }
      
  div.module {
  padding-bottom:1.5em;
  padding-top:1.5em;
  }
	  
  div.footer {
  background-color: #EEE;
  border:0;
  margin:0;
  padding: 0.5em;
  font-size: 100%;
  font-weight: bold;
  width:100%;
  }


  a {
  color: #000080;
  }

  a:hover {
  color: #800000;
  }
      
  h2 {
  color: #800000;
  padding-bottom: 0;
  margin-bottom: 0;
  clear:left;
  }

  table { 
  margin-left: 3em;
  text-align: center;
  }
  
  th { 
  text-align: center;
  background-color: #000080;
  color: #FFF;
  padding: 0.4em;
  }      
  
  td { 
  font-family: monospace; 
  text-align: left;
  background-color: #EEE;
  color: #000;
  padding: 0.4em;
  }

  img {
  padding-top: 0;
  margin-top: 0;
  border-top: 0;
  }

  
  p {
  padding-top: 0;
  margin-top: 0;
  }
  
</style>

</head>
<body>
<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Wed 15 Oct 2014<br />
RNAseq.fastq
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/warning.png" alt="[WARNING]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>RNAseq.fastq</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>153557</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>101</td>
</tr>
<tr>
<td>%GC</td>
<td>46</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/tick.png" alt="[OK]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/warning.png" alt="[WARN]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/warning.png" alt="[WARN]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>GTTTATTTGAATTTTTGGTATCTTTAAAGCTGAACTTAGATTCATTTTCT</td>
<td>315</td>
<td>0.2051355522704924</td>
<td>No Hit</td>
</tr>
<tr>
<td>CTAAAATTCATTTTGTTAAGTCTTATACTAGTTAATTCATTATGCAAAAG</td>
<td>202</td>
<td>0.13154724304329987</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/warning.png" alt="[WARN]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>AAAAA</td>
<td>63625</td>
<td>3.0921767</td>
<td>9.875067</td>
<td>2</td>
</tr>
<tr>
<td>TTTTT</td>
<td>60400</td>
<td>3.0083098</td>
<td>5.651855</td>
<td>2</td>
</tr>
<tr>
<td>CTTCT</td>
<td>41320</td>
<td>2.6783879</td>
<td>5.59602</td>
<td>5</td>
</tr>
<tr>
<td>CTCCT</td>
<td>35065</td>
<td>2.5929892</td>
<td>5.812241</td>
<td>8</td>
</tr>
<tr>
<td>TCTTC</td>
<td>37445</td>
<td>2.4272077</td>
<td>5.6923738</td>
<td>7</td>
</tr>
<tr>
<td>GAAGA</td>
<td>37125</td>
<td>2.4145603</td>
<td>5.2671733</td>
<td>2</td>
</tr>
<tr>
<td>AAGAA</td>
<td>42395</td>
<td>2.3835194</td>
<td>5.4255857</td>
<td>2</td>
</tr>
<tr>
<td>CAGCT</td>
<td>31635</td>
<td>2.3490317</td>
<td>5.329326</td>
<td>3</td>
</tr>
<tr>
<td>AGAAA</td>
<td>40995</td>
<td>2.3048089</td>
<td>7.0069127</td>
<td>2</td>
</tr>
<tr>
<td>CCAGG</td>
<td>26640</td>
<td>2.2771568</td>
<td>5.3499556</td>
<td>8</td>
</tr>
<tr>
<td>CTGGA</td>
<td>30105</td>
<td>2.2557118</td>
<td>6.721901</td>
<td>1</td>
</tr>
<tr>
<td>TCCAG</td>
<td>29450</td>
<td>2.1867864</td>
<td>8.934587</td>
<td>7</td>
</tr>
<tr>
<td>CTCCA</td>
<td>29055</td>
<td>2.1380503</td>
<td>7.7474446</td>
<td>6</td>
</tr>
<tr>
<td>CTGGG</td>
<td>24490</td>
<td>2.1227622</td>
<td>6.6832094</td>
<td>1</td>
</tr>
<tr>
<td>CTTCA</td>
<td>30525</td>
<td>1.9689695</td>
<td>6.599958</td>
<td>1</td>
</tr>
<tr>
<td>AAAGA</td>
<td>34080</td>
<td>1.9160359</td>
<td>5.4255857</td>
<td>2</td>
</tr>
<tr>
<td>TCCTG</td>
<td>25380</td>
<td>1.8938365</td>
<td>5.1409206</td>
<td>7</td>
</tr>
<tr>
<td>TCTCC</td>
<td>25445</td>
<td>1.8816088</td>
<td>6.422168</td>
<td>7</td>
</tr>
<tr>
<td>CTGGC</td>
<td>21885</td>
<td>1.8799015</td>
<td>5.165182</td>
<td>1</td>
</tr>
<tr>
<td>GCTCC</td>
<td>22055</td>
<td>1.8774642</td>
<td>5.6149983</td>
<td>5</td>
</tr>
<tr>
<td>GAAAA</td>
<td>33300</td>
<td>1.872183</td>
<td>5.5889893</td>
<td>1</td>
</tr>
<tr>
<td>CAGAA</td>
<td>28030</td>
<td>1.8066363</td>
<td>5.4071593</td>
<td>1</td>
</tr>
<tr>
<td>AAAAT</td>
<td>36660</td>
<td>1.7904359</td>
<td>7.934155</td>
<td>3</td>
</tr>
<tr>
<td>CTTGG</td>
<td>22485</td>
<td>1.6930419</td>
<td>6.2072487</td>
<td>1</td>
</tr>
<tr>
<td>CTCAG</td>
<td>22780</td>
<td>1.6915108</td>
<td>8.245783</td>
<td>1</td>
</tr>
<tr>
<td>CCACC</td>
<td>20030</td>
<td>1.6814799</td>
<td>5.9465537</td>
<td>8</td>
</tr>
<tr>
<td>CTGAA</td>
<td>25860</td>
<td>1.674966</td>
<td>6.784324</td>
<td>1</td>
</tr>
<tr>
<td>TCTGC</td>
<td>21900</td>
<td>1.6341615</td>
<td>5.937402</td>
<td>7</td>
</tr>
<tr>
<td>CTGAG</td>
<td>21595</td>
<td>1.6180733</td>
<td>5.813536</td>
<td>1</td>
</tr>
<tr>
<td>TTCAT</td>
<td>28175</td>
<td>1.5930649</td>
<td>5.3494096</td>
<td>7</td>
</tr>
<tr>
<td>CAAAA</td>
<td>28130</td>
<td>1.5672916</td>
<td>5.160465</td>
<td>1</td>
</tr>
<tr>
<td>TTCCA</td>
<td>22805</td>
<td>1.4710022</td>
<td>5.508047</td>
<td>6</td>
</tr>
<tr>
<td>AAGAT</td>
<td>24520</td>
<td>1.3853334</td>
<td>6.54819</td>
<td>3</td>
</tr>
<tr>
<td>AAAAC</td>
<td>24745</td>
<td>1.3786929</td>
<td>5.160633</td>
<td>3</td>
</tr>
<tr>
<td>CTTGA</td>
<td>21065</td>
<td>1.3710988</td>
<td>6.5336075</td>
<td>1</td>
</tr>
<tr>
<td>TCCAC</td>
<td>18575</td>
<td>1.3668659</td>
<td>6.7120724</td>
<td>7</td>
</tr>
<tr>
<td>AAATT</td>
<td>27720</td>
<td>1.3604711</td>
<td>5.4741096</td>
<td>4</td>
</tr>
<tr>
<td>GTTTT</td>
<td>22530</td>
<td>1.2917682</td>
<td>5.22705</td>
<td>1</td>
</tr>
<tr>
<td>GAAAT</td>
<td>22500</td>
<td>1.2712073</td>
<td>5.58925</td>
<td>3</td>
</tr>
<tr>
<td>TTTGA</td>
<td>21975</td>
<td>1.253783</td>
<td>5.3425984</td>
<td>6</td>
</tr>
<tr>
<td>TTGAA</td>
<td>21405</td>
<td>1.2152871</td>
<td>5.8398438</td>
<td>7</td>
</tr>
<tr>
<td>ATTCA</td>
<td>20125</td>
<td>1.1323367</td>
<td>6.1694984</td>
<td>6</td>
</tr>
<tr>
<td>ATCCA</td>
<td>17545</td>
<td>1.1261774</td>
<td>7.629943</td>
<td>6</td>
</tr>
<tr>
<td>TCCAA</td>
<td>17460</td>
<td>1.1207216</td>
<td>5.1385336</td>
<td>7</td>
</tr>
<tr>
<td>AATCT</td>
<td>19810</td>
<td>1.1146133</td>
<td>5.2394667</td>
<td>5</td>
</tr>
<tr>
<td>AAATC</td>
<td>19140</td>
<td>1.0716472</td>
<td>5.4032183</td>
<td>4</td>
</tr>
<tr>
<td>AGATC</td>
<td>15785</td>
<td>1.0224029</td>
<td>6.062117</td>
<td>4</td>
</tr>
<tr>
<td>AATCC</td>
<td>14250</td>
<td>0.91467816</td>
<td>5.4480066</td>
<td>5</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
</body></html>