<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
		  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns='http://www.w3.org/1999/xhtml' xml:lang='en' lang='en'> 
<head>
  <meta http-equiv="Content-Type" 
		content="text/html; charset=utf-8"></meta>
  <link rel="stylesheet" type="text/css" href="QA.css" />
  <title>ShortRead Quality Assessment</title>
</head>
<body>
  <div id="header">
	<div id="logo"><a href="http://bioconductor.org">
		<img src="./image/bioclogo-small.gif" alt="Bioconductor" />
	</a></div>
	<div id="locationline"><p><ul id="pagelocation">
		<li><em>ShortRead</em> Quality Assessment</li>
	</ul></p></div>
  </div>

<h2>Overview</h2>
<a name="Overview"></a>

<p>
  This document provides a quality assessment of Genome Analyzer
  results. The assessment is meant to complement, rather than replace,
  quality assessment available from the Genome Analyzer and its
  documentation. The narrative interpretation is based on experience
  of the package maintainer. It is applicable to results from the
  'Genome Analyzer' hardware single-end module, configured to scan 300
  tiles per lane. The 'control' results refered to below are from
  analysis of PhiX-174 sequence provided by Illumina.
</p>
<h2>Run Summary</h2>
<a name="Run-Summary"></a>

<p>
  Subsequent sections of the report use the following to identify figures and
  other information.
</p>
<table border="0">
<tr>
<td></td><td>Key</td></tr>
<tr>
<td>trimmedRNA.fastq</td><td>1</td></tr>
</table>


<p>
  Read counts. Filtered and aligned read counts are reported relative
  to the total number of reads (clusters; if only filtered or aligned
  reads are available, total read count is reported). Consult Genome
  Analyzer documentation for official guidelines. From experience,
  very good runs of the Genome Analyzer 'control' lane result in 25-30
  million reads, with up to 95&#37; passing pre-defined filters.
</p>
<pre>
  ShortRead:::.ppnCount(qa[["readCounts"]])
</pre>
<table border="0">
<tr>
<td></td><td>read</td><td>filter</td><td>aligned</td></tr>
<tr>
<td>1</td><td>112364</td><td>&nbsp;</td><td>&nbsp;</td></tr>
</table>

<pre>
  ShortRead:::.plotReadCount(qa)
</pre>
<a href="./image/readCount.pdf"><img border="0" src="./image/readCount.jpg" alt="./image/readCount.jpg"></img></a>

<p>
  Base call frequency over all reads. Base frequencies should accurately
  reflect the frequencies of the regions sequenced.
</p>
<pre>
  ShortRead:::.plotNucleotideCount(qa)
</pre> 
<a href="./image/baseCalls.pdf"><img border="0" src="./image/baseCalls.jpg" alt="./image/baseCalls.jpg"></img></a>

<p>
  Overall read quality. Lanes with consistently good quality reads have
  strong peaks at the right of the panel.
</p>
<pre>
  df &lt;- qa[["readQualityScore"]]
  ShortRead:::.plotReadQuality(df[df$type=="read",])
</pre>
<a href="./image/readQuality.pdf"><img border="0" src="./image/readQuality.jpg" alt="./image/readQuality.jpg"></img></a>
<h2>Read Distribution</h2>
<a name="Read-Distribution"></a>

<p>
  These curves show how coverage is distributed amongst
  reads. Ideally, the cumulative proportion of reads will transition
  sharply from low to high.
</p>
<p>
  Portions to the left of the transition might correspond roughly to
  sequencing or sample processing errors, and correspond to reads that
  are represented relatively infrequently. 10-15&#37;; of reads in a
  typical Genome Analyzer 'control' lane fall in this category.
</p>
<p>
  Portions to the right of the transition represent reads that are
  over-represented compared to expectation. These might include
  inadvertently sequenced primer or adapter sequences, sequencing or
  base calling artifacts (e.g., poly-A reads), or features of the
  sample DNA (highly repeated regions) not adequately removed during
  sample preparation. About 5&#37; of Genome Analyzer 'control' lane
  reads fall in this category.
</p>
<p>
  Broad transitions from low to high cumulative proportion of reads
  may reflect sequencing bias or (perhaps intentional) features of
  sample preparation resulting in non-uniform coverage. the transition
  is about 5 times as wide as expected from uniform sampling across
  the Genome Analyzer 'control' lane.
</p>
<pre>
  df &lt;- qa[["sequenceDistribution"]]
  ShortRead:::.plotReadOccurrences(df[df$type=="read",], cex=.5)
</pre>
<a href="./image/readOccurences.pdf"><img border="0" src="./image/readOccurences.jpg" alt="./image/readOccurences.jpg"></img></a>

<p>
  Common duplicate reads might provide clues to the source of
  over-represented sequences. Some of these reads are filtered by the
  alignment algorithms; other duplicate reads might point to sample
  preparation issues.
</p>
<pre>
  ShortRead:::.freqSequences(qa, "read")
</pre>
<table border="0">
<tr>
<td>sequence</td><td>count</td><td>lane</td></tr>
<tr>
<td>TCTTTAAAGCTGAACTTAGATTCATTTTCTGGACAACCAGCTATCACCAGGCTCGTTAGGCATGTCACCTCTACTTACAA</td><td>255</td><td>1</td></tr>
<tr>
<td>TCTTATACTAGTTAATTCATTATGCAAAAGGTACAAGGGGTAATCTTTGCTTTTTTATACTTTTAGTTTTTTCTTTCATC</td><td>161</td><td>1</td></tr>
<tr>
<td>AAAGCAAAGATTACCCCTTGTACCTTTTGCATAATGAATTAACTAGTATAAGACTTAACAAAATGAATTTTAGCTAAGCA</td><td> 85</td><td>1</td></tr>
<tr>
<td>AACCAGCTATCACCAGGCTCGTTAGGCATGTCACCTCTACTTACAAATCTTCTCACTATTTTGCCACATAGATGAGTTAG</td><td> 79</td><td>1</td></tr>
<tr>
<td>TTAATCTTTCCTTAGATGCACTCCTGTGTTGGGTTAACAGTATAATCAATAAATTGTTTATTGTTGGATTATTTTTATTG</td><td> 79</td><td>1</td></tr>
<tr>
<td>TAAAATTCATTTTGTTAAGTCTTATACTAGTTAATTCATTATGCAAAAGGTACAAGGGGTAATCTTTGCTTTTTTATACT</td><td> 67</td><td>1</td></tr>
<tr>
<td>ACTGTAGCTTTAAAAGTTAGTCTAAAAAGGTACAGCCTTTTAGAAACGGATACAACCTTGACTAGAGAGTAAAATTTAAC</td><td> 65</td><td>1</td></tr>
<tr>
<td>CGCAAGGGAACGATGAAAGAAAAAACTAAAAGTATAAAAAAGCAAAGATTACCCCTTGTACCTTTTGCATAATGAATTAA</td><td> 63</td><td>1</td></tr>
<tr>
<td>TATAGAGAAAGTACCGCAAGGGAACGATGAAAGAAAAAACTAAAAGTATAAAAAAGCAAAGATTACCCCTTGTACCTTTT</td><td> 61</td><td>1</td></tr>
<tr>
<td>GTTGTTGAGCTTTAACGCTTTCTTAATTGATGGCTGCTTTTAGGCCTACTATGGTAGTGTTAAATTTTACTCTCTAGTCA</td><td> 57</td><td>1</td></tr>
<tr>
<td>TCTTTCCTTAGATGCACTCCTGTGTTGGGTTAACAGTATAATCAATAAATTGTTTATTGTTGGATTATTTTTATTGACTG</td><td> 55</td><td>1</td></tr>
<tr>
<td>AGAATAAAACAAAACATTTAATCCCAATTTAAAGTATAGGAGATAGAAATCTAAGTACGGCGCTATAGAGAAAGTACCGC</td><td> 54</td><td>1</td></tr>
<tr>
<td>TGAACTTAGATTCATTTTCTGGACAACCAGCTATCACCAGGCTCGTTAGGCATGTCACCTCTACTTACAAATCTTCTCAC</td><td> 53</td><td>1</td></tr>
<tr>
<td>ACCAAAAATTCAAATAAACCCCACTGTAGCTTTAAAAGTTAGTCTAAAAAGGTACAGCCTTTTAGAAACGGATACAACCT</td><td> 52</td><td>1</td></tr>
<tr>
<td>GTTAAATTTTACTCTCTAGTCAAGGTTGTATCCGTTTCTAAAAGGCTGTACCTTTTTAGACTAACTTTTAAAGCTACAGT</td><td> 52</td><td>1</td></tr>
<tr>
<td>AACAACCAAGATAGAATAAAACAAAACATTTAATCCCAATTTAAAGTATAGGAGATAGAAATCTAAGTACGGCGCTATAG</td><td> 49</td><td>1</td></tr>
<tr>
<td>TAGCTAAGACCCAAACTGGGATTAGATACCCCACTATGCTTAGCCCTAAACACAGATAATTACATAAACAAAATTATTCG</td><td> 46</td><td>1</td></tr>
<tr>
<td>TTTCTTTCATCGTTCCCTTGCGGTACTTTCTCTATAGCGCCGTACTTAGATTTCTATCTCCTATACTTTAAATTGGGATT</td><td> 45</td><td>1</td></tr>
<tr>
<td>AGCTTTAAAGATACCAAAAATTCAAATAAACCCCACTGTAGCTTTAAAAGTTAGTCTAAAAAGGTACAGCCTTTTAGAAA</td><td> 44</td><td>1</td></tr>
<tr>
<td>AACAAAACATTTAATCCCAATTTAAAGTATAGGAGATAGAAATCTAAGTACGGCGCTATAGAGAAAGTACCGCAAGGGAA</td><td> 40</td><td>1</td></tr>
</table>


<p>
  Common duplicate reads after filtering
</p>
<pre>
  ShortRead:::.freqSequences(qa, "filtered")
</pre>
<pre>NA</pre>

<p>
  Common aligned duplicate reads are
</p>
<pre>
  ShortRead:::.freqSequences(qa, "aligned")
</pre>
<pre>NA</pre>
<h2>Cycle-Specific Base Calls and Read Quality</h2>
<a name="Cycle-Specific-Base-Calls-and-Read-Quality"></a>

<p>
Per-cycle base call should usually be approximately uniform across
cycles. Genome Analyzer `control' lane results often show a deline in
A and increase in T as cycles progress. This is likely an artifact of
the underlying technology.
</p>
<pre>
  perCycle &lt;- qa[["perCycle"]]
  ShortRead:::.plotCycleBaseCall(perCycle$baseCall)
</pre>
<a href="./image/perCycleBaseCall.pdf"><img border="0" src="./image/perCycleBaseCall.jpg" alt="./image/perCycleBaseCall.jpg"></img></a>

<p>
Per-cycle quality score. Reported quality scores are `calibrated',
i.e., incorporating phred-like adjustments following sequence
alignment. These typically decline with cycle, in an accelerating
manner. Abrupt transitions in quality between cycles toward the end of
the read might result when only some of the cycles are used for
alignment: the cycles included in the alignment are calibrated more
effectively than the reads excluded from the alignment.
</p>
<p>
The reddish lines are quartiles (solid: median, dotted: 25, 75), the green line
is the mean. Shading is proporitional to number of reads.
</p>
<pre>
  perCycle &lt;- qa[["perCycle"]]
  ShortRead:::.plotCycleQuality(perCycle$quality)
</pre>
<a href="./image/perCycleQuality.pdf"><img border="0" src="./image/perCycleQuality.jpg" alt="./image/perCycleQuality.jpg"></img></a>
<h2>Adapter Contamination</h2>
<a name="Adapter-Contamination"></a>

<p>
 Adapter contamination is defined here as 
 non-genetic sequences attached at either or both ends 
 of the reads. The 'contamination' measure is
 the number of reads with a right or left match to the adapter
 sequence over the total number of reads. Mismatch rates
 are 10&#37; on the left and 20&#37; on the right with a
 minimum overlap of 10 nt. 
</p>
<pre>
  ShortRead:::.ppnCount(qa[["adapterContamination"]])
</pre>
<table border="0">
<tr>
<td></td><td>contamination</td></tr>
<tr>
<td>1</td><td>1</td></tr>
</table>


  <hr />
  <p>
	Wed Oct 15 17:35:03 2014; ShortRead v. 1.22.0<br />
	Report template: Martin Morgan
  </p>
</body>
</html>
